queries/wit/highlight.scm

```diff
@@ -0,0 +1,96 @@
+; Comments
+(line_comment) @comment
+
+(block_comment) @comment
+
+; Primitive Types
+[
+  "bool"
+  "s8"
+  "s16"
+  "s32"
+  "s64"
+  "u8"
+  "u16"
+  "u32"
+  "u64"
+  "float32"
+  "float64"
+  "char"
+  "string"
+] @type.builtin
+
+; Container Types
+[
+  "list"
+  "tuple"
+  "option"
+  "result"
+] @type.builtin
+
+"func" @keyword.function
+
+; Keywords for file structure and components
+[
+  "record"
+  "enum"
+  "variant"
+  "flags"
+  "resource"
+] @keyword
+
+; Keywords for importing and exporting
+[
+  "package"
+  ; "include" TODO: Implement me
+  "world"
+  "use"
+  "import"
+  ; "export" TODO: Implement me
+] @keyword.import
+
+; Resource Keywords
+[
+  ; "constructor" TODO: Implement me
+  "static"
+  ; "self" TODO: Implement me
+  ; "borrow" TODO: Implement me
+  ; "owned"
+] @keyword.modifier
+
+; Identifiers (variable names, function names, type names, etc.)
+((identifier) @variable
+  (#match? @variable "^[a-z_][a-zA-Z0-9_]*$"))
+
+; Constants (UPPER_CASE names and Enums)
+((identifier) @constant
+  (#match? @constant "^[A-Z][A-Z0-9_]+$"))
+
+; Functions and Methods (lowercase names followed by parentheses)
+((identifier) @function
+  (#match? @function "^[a-z_][a-zA-Z0-9_]*\\("))
+
+; Named Types (Capitalized identifiers)
+((identifier) @type
+  (#match? @type "^[A-Z]"))
```

Comment:
All of these patterns should use `lua-match` instead of `match`. `\\(` -> `%(` for example

Reply:
I updated it to be like so, I hope this is what you intended? Please resolve if yes 😊

```scm
; Identifiers (variable names, function names, type names, etc.)
((identifier) @variable
 (#lua-match? @variable "string.match(v, '^[a-z_][a-zA-Z0-9_]*$')"))

; Constants (UPPER_CASE names and Enums)
((identifier) @constant
 (#lua-match? @constant "string.match(v, '^[A-Z][A-Z0-9_]+$')"))

; Functions and Methods (lowercase names followed by parentheses)
((identifier) @function
 (#lua-match? @function "string.match(v, '^[a-z_][a-zA-Z0-9_]*%(')"))

; Named Types (Capitalized identifiers)
((identifier) @type
 (#lua-match? @type "string.match(v, '^[A-Z]')"))
```

Reply:
no no no. Just the lua pattern. No need for the entire lua expression for matching

Reply:
Ok thanks, hopefully it is correct now.